tower/src/util/clone_boxed.rs

```diff
@@ -0,0 +1,138 @@
+use super::ServiceExt;
+use futures_util::future::BoxFuture;
+use std::{
+    fmt,
+    task::{Context, Poll},
+};
+use tower_layer::{layer_fn, LayerFn};
+use tower_service::Service;
+
+/// A `Clone + Send` boxed `Service`.
+///
+/// [`CloneBoxService`] turns a service into a trait object, allowing the
+/// response future type to be dynamic, and allowing the service to be cloned.
```

Comment:
it would be nice if the docs for this included a comparison with the non-`Clone` `BoxService` type? something like "This is similar to the [`BoxService`] type, but ..."